static/js/app.ts

```diff
@@ -218,7 +220,33 @@ export function runit(level: string, lang: string, cb: () => void) {
       }
       if (response.Error) {
         error.show(ErrorMessages['Transpile_error'], response.Error);
-        if (response.Location && response.Location[0] != "?") {
+        if (response.Invalid_command)  {
```

Comment:
With the new generic `LocationLength` field:

```suggestion
        if (response.Location && response.Location[0] != '?' && response.LocationLength)  {
```

Probably, there is now a bunch of duplication between the "complete line" error marker and the "specific" error marker blocks.

So probably the logic should look something like this:

```ts
        if (response.Location && response.Location[0] != '?') {
            // Add (line) annotation

            if (response.LocationLength) {
                // Add (range) marker
            }
        }
```